tests/PHPStan/Parser/PhpDocParserTest.php

```diff
@@ -3480,6 +3482,91 @@ public function provideTypeAliasImportTagsData(): Iterator
 		];
 	}
 
+	public function provideAssertTagsData(): Iterator
+	{
+		yield [
+			'OK',
+			'/** @phpstan-assert Type $var */',
+			new PhpDocNode([
+				new PhpDocTagNode(
+					'@phpstan-assert',
+					new AssertTagValueNode(
+						new IdentifierTypeNode('Type'),
+						'$var',
+						''
+					)
+				),
+			]),
+		];
+
+		yield [
+			'OK with psalm syntax',
+			'/** @psalm-assert Type $var */',
+			new PhpDocNode([
+				new PhpDocTagNode(
+					'@psalm-assert',
+					new AssertTagValueNode(
+						new IdentifierTypeNode('Type'),
+						'$var',
+						''
+					)
+				),
+			]),
+		];
+
+		yield [
+			'OK with description',
+			'/** @phpstan-assert Type $var assert Type to $var */',
+			new PhpDocNode([
+				new PhpDocTagNode(
+					'@phpstan-assert',
+					new AssertTagValueNode(
+						new IdentifierTypeNode('Type'),
+						'$var',
+						'assert Type to $var'
+					)
+				),
+			]),
+		];
+
+		yield [
+			'OK with union type',
+			'/** @phpstan-assert Type|Other $var */',
+			new PhpDocNode([
+				new PhpDocTagNode(
+					'@phpstan-assert',
+					new AssertTagValueNode(
+						new UnionTypeNode([
+							new IdentifierTypeNode('Type'),
+							new IdentifierTypeNode('Other'),
+						]),
+						'$var',
+						''
+					)
+				),
+			]),
+		];
+
+		yield [
+			'invalid $this->method()',
+			'/** @phpstan-assert Type $this->method() */',
```

Comment:
This would be a nice future addition

Reply:
Does this mean the current implementation doesn't support class props (e.g. `$this->myProp`), or are you only talking about methods?

Reply:
Any syntax other than plain variables hasn't been implemented yet. It will definitely happen in the future.